runtime/tests/shared/SharedCacheAPITest.cpp

```diff
@@ -59,7 +59,7 @@ getCacheDir(J9JavaVM *vm, char *cacheDir)
 		j9tty_printf(PORTLIB, "Cannot get a directory\n");
 	}
 
-	sprintf(cacheDir, "%s%s", cacheDirNoTestBasedir, TEST_BASEDIR);
+	snprintf(cacheDir, bufLen, "%s%s", cacheDirNoTestBasedir, TEST_BASEDIR);
```

Comment:
A more robust solution would pass a smaller length to `j9shmem_getDir()` so there would be no possibility of overflow adding `TEST_BASEDIR` (and eliminate the need for the second buffer `cacheDirNoTestBasedir`):
```cpp
void
getCacheDir(J9JavaVM *vm, char *cacheDir, size_t bufLen)
{
	PORT_ACCESS_FROM_JAVAVM(vm);

	cacheDir[0] = '\0';

	if (bufLen < LITERAL_STRLEN(TEST_BASEDIR)) {
		j9tty_printf(PORTLIB, "Buffer too small to get directory\n");
	} else {
		U_32 flags = J9SHMEM_GETDIR_APPEND_BASEDIR
#if defined(OPENJ9_BUILD) && !defined(J9ZOS390)
				| J9SHMEM_GETDIR_USE_USERHOME
#endif /* defined(OPENJ9_BUILD) && !defined(J9ZOS390) */
				;
		IDATA rc = j9shmem_getDir(NULL, flags, cacheDir, bufLen - LITERAL_STRLEN(TEST_BASEDIR));
		if (rc < 0) {
			j9tty_printf(PORTLIB, "Cannot get a directory\n");
		} else {
			strcat(cacheDir, TEST_BASEDIR);
		}
	}
}
```

Reply:
Thank you.
I updated `getCacheDir()` as suggested, with one minor change:
`if (bufLen < LITERAL_STRLEN(TEST_BASEDIR))` -> `if (bufLen <= LITERAL_STRLEN(TEST_BASEDIR))`